webapp/channels/src/components/admin_console/custom_plugin_settings/index.ts

```diff
@@ -132,16 +132,33 @@ function makeGetPluginSchema() {
             }
 
             if (plugin.id !== appsPluginID || appsFeatureFlagIsEnabled) {
-                const pluginEnableSetting = getEnablePluginSetting(plugin);
-                if (pluginEnableSetting.isDisabled) {
-                    pluginEnableSetting.isDisabled = it.any(pluginEnableSetting.isDisabled, it.not(it.userHasWritePermissionOnResource('plugins')));
-                } else {
-                    pluginEnableSetting.isDisabled = it.not(it.userHasWritePermissionOnResource('plugins'));
-                }
+                const pluginEnableSetting = getEnablePluginSetting(plugin) as AdminDefinitionSetting;
+
+                if (sections.length > 0 && sections.filter((s) => (s.settings[0] as AdminDefinitionSettingBanner).banner_type !== 'warning').length === 0) {
+                    // If the plugin is composed of purely custom sections (e.g. Calls) and it's disabled, we show a single warning.
```

Comment:
What's the difference between a custom section and a regular one? Unlike with individual settings, I would've expected that the basic styling of sections would be pretty consistent between plugins

Reply:
@hmhealey In the case of Calls we need to render some extra stuff like:

- Enterprise labels 
- More complex components that include an upsell flow (button + modal)

![image](https://github.com/user-attachments/assets/9ed5f0df-0571-470b-b426-e2570d77deb1)

Reply:
Ah, I see. I could see making the enterprise badge into an extra prop for the existing section or there being a more generic `titleBadge` attribute. I'm not sure about having the button open a modal, but I would think that that could be a custom setting in the section. Alternately, since we have multiple plugins with Enterprise elements, perhaps we could even have some sort of `isEnterprise` attribute which adds the badge and button automatically so that it doesn't need to be part of each plugin.

That's all just brainstorming though